cdap-proto/src/main/java/io/cdap/cdap/proto/id/EntityId.java

```diff
@@ -120,7 +120,8 @@ public static boolean isValidVersionId(String datasetId) {
   }
 
   private final EntityType entity;
-  private Vector<EntityId> hierarchy;
+  // Hierarchy should be transient to avoid recursive JSON serialization.
```

Comment:
nit: extend the comment explaining this field is for caching the result of hierarchical traversal, caller should always  access this via get accessor() and never via direct access to this variable.

Reply:
Since this is a private variable, a caller won't be able to directly access the variable. However, I'll modify the comment to add the caching detail for readability.